### PR TITLE
PPC64 big endian

### PR DESCRIPTION
**Pull request by nathanchance**
This is a respin of #90 + @shenki's https://github.com/shenki/continuous-integration/commits/powerpc64.

As noted, we can work around https://github.com/ClangBuiltLinux/linux/issues/308 by just booting the raw kernel image like we do for MIPS.

Presubmit: https://travis-ci.com/nathanchance/continuous-integration/builds/121660194

### CODE REVIEW

File: driver.sh [hunk 1]
```diff
@@ -288,8 +299,8 @@ build_linux() {
     cat ../configs/common.config >> .config
     # Some torture test configs cause issues on x86_64
     [[ $ARCH != "x86_64" ]] && cat ../configs/tt.config >> .config
-    # Disable ftrace on arm32: https://github.com/ClangBuiltLinux/linux/issues/35
-    [[ $ARCH == "arm" ]] && ./scripts/config -d CONFIG_FTRACE
+    # Disable ftrace on arm32 and ppc64be: https://github.com/ClangBuiltLinux/linux/issues/35
+    [[ $ARCH == "arm" || $SUBARCH == "ppc64" ]] && ./scripts/config -d CONFIG_FTRACE
```
Comment (nickdesaulniers) [inline driver.sh]: what's wrong w/ ftrace on ppc64?

File: patches/llvm-all/linux/ppc64/0001-powerpc-vdso32-Drop-mabi-elfv1-for-32-bit-objects.patch [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+Subject: [PATCH] powerpc/vdso32: Drop -mabi=elfv1 for 32 bit objects
+
+All 64-bit objects need to specify the flag to be compiled correctly, we
+just don't need it for 32-bit objects. GCC just ignored it, but clang
+doesn't.
+
+Link: https://github.com/ClangBuiltLinux/linux/issues/240
+Signed-off-by: Daniel Axtens <dja@axtens.net>
+Signed-off-by: Joel Stanley <joel@jms.id.au>
+(am from https://lore.kernel.org/linuxppc-dev/20190110014258.20963-1-joel@jms.id.au/)
```
Comment (nickdesaulniers) [inline patches/llvm-all/linux/ppc64/0001-powerpc-vdso32-Drop-mabi-elfv1-for-32-bit-objects.patch]: @daxtens send a v2 plz 📉

**Comment by daxtens** [comment]
ah yikes, will try to get to that.

Regards,
Daniel

>

File: driver.sh (hunk 1 above)
Comment (nathanchance) [reply driver.sh]: It was broken at one point...

https://travis-ci.org/shenki/continuous-integration/jobs/474371134

Looks like it works now, I am bisecting the commit that fixes it just for completeness.

https://travis-ci.com/nathanchance/continuous-integration/builds/122211446

**Pull request merged**